src/iplant_groups/clients/grouper.clj

```diff
@@ -484,32 +484,42 @@
 
 ;; This is only available as a Lite request; ActAsSubject works differently.
 (defn- format-group-folder-privileges-lookup-request
-  [entity-type username group-or-folder-name]
-  (if-let [name-key (get {:group :groupName
+  [entity-type username group-or-folder-name params]
+  (if-let [name-key (get {:group  :groupName
                           :folder :stemName}
                          entity-type)]
     {:WsRestGetGrouperPrivilegesLiteRequest
-     {:actAsSubjectId username
-      :includeSubjectDetail "T"
-      :includeGroupDetail   "T"
-      name-key group-or-folder-name}}
+     (remove-vals nil? {:actAsSubjectId       username
+                        :includeSubjectDetail "T"
+                        :includeGroupDetail   "T"
+                        name-key              group-or-folder-name
+                        :subjectId            (:subject-id params)
+                        :subjectSourceId      (:subject-source-id params)
+                        :privilegeName        (:privilege params)})}
     (throw+ {:type :clojure-commons.exception/bad-request :entity-type entity-type})))
 
+(defn- filter-privileges
+  [privileges {:keys [subject-source-id]}]
+  (if subject-source-id
+    (filter (fn [priv] (= (:sourceId (:wsSubject priv)) subject-source-id)) privileges)
+    privileges))
```

Comment:
Grouper only applies the subject source ID to the filter if the subject ID is also provided. I wanted to be able to filter by subject source ID alone, so I added this.